common/src/main/java/io/druid/common/utils/SerializerUtils.java

```diff
@@ -38,6 +38,32 @@
 {
   private static final Charset UTF8 = Charset.forName("UTF-8");
 
+  /**
+   * Writes the given long value into the given OutputStream, using the helperBuffer. Faster alternative to
+   * out.write(Longs.toByteArray(value)), more convenient (sometimes) than wrapping the OutputStream into {@link
+   * java.io.DataOutputStream}.
+   *
+   * @param helperBuffer must be a big-endian, heap ByteBuffer of capacity 8.
+   */
+  public static void writeLongToOutputStream(OutputStream out, long value, ByteBuffer helperBuffer) throws IOException
+  {
+    helperBuffer.putLong(0, value);
```

Comment:
Ok, it seems that it is used not only with big endian buffers, but also with little-endian buffers in `GenericIndexedWriter.initializeHeaderOutLong()`. So I removed checks and "requirement" for helperBuffer to be big-endian from javadocs